packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/FieldSelect.tsx

```diff
@@ -0,0 +1,45 @@
+import { Field, getItemId, TableCalculation } from '@lightdash/common';
+import { Select } from '@mantine/core';
+import { FC } from 'react';
+import FieldIcon from '../../common/Filters/FieldIcon';
+import { fieldLabelText } from '../../common/Filters/FieldLabel';
+import FieldSelectItem from '../FieldSelectItem';
+
+interface Props {
+    selectedField?: Field | TableCalculation;
+    fieldOptions: (Field | TableCalculation)[];
+    placeholder?: string;
+    onChange: (newValue: string | null) => void;
+}
+
+const FieldSelect: FC<Props> = ({
```

Comment:
(not in the scope of this PR) feels like we are recreating this for every single config we have and we've only been reusing `FieldSelectItem`. I believe there's one one in pie charts/series and table/condit-formatting

Reply:
I refactored it here, but I have a ticket to use it everywhere else. I'hope to get to that soon.